packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -476,15 +476,18 @@ class ThemeData with Diagnosticable {
     assert(colorSchemeSeed == null || primaryColor == null);
     final Brightness effectiveBrightness = brightness ?? colorScheme?.brightness ?? Brightness.light;
     final bool isDark = effectiveBrightness == Brightness.dark;
-    if (colorSchemeSeed != null) {
-      colorScheme = ColorScheme.fromSeed(seedColor: colorSchemeSeed, brightness: effectiveBrightness);
+    if (colorSchemeSeed != null || useMaterial3) {
+      if (colorSchemeSeed != null) {
+        colorScheme = ColorScheme.fromSeed(seedColor: colorSchemeSeed, brightness: effectiveBrightness);
+      }
+      colorScheme ??= isDark ? _colorSchemeDarkM3 : _colorSchemeLightM3;
 
       // For surfaces that use primary color in light themes and surface color in dark
       final Color primarySurfaceColor = isDark ? colorScheme.surface : colorScheme.primary;
       final Color onPrimarySurfaceColor = isDark ? colorScheme.onSurface : colorScheme.onPrimary;
 
       // Default some of the color settings to values from the color scheme
-      primaryColor = primarySurfaceColor;
+      primaryColor ??= primarySurfaceColor;
```

Comment:
This is a little tricky here. We are setting up the color scheme as needed, but then trying to patch ThemeData colors to match the scheme as best we can. This got us into a[ bit of trouble](https://github.com/flutter/flutter/issues/107105) as this only happens if the ThemeData constructs the color scheme and not if the color scheme is created and passed to the constructor. I wonder if we should try to address that with this PR, as it will already be changing the default colors for M3.

Reply:
I see. This change is to fix the failing test in G3. Some apps have already set useMaterial3 to be true and used the primaryColor for Image color in the dark mode, which caused some missing images in the apps after I used the default colorScheme. Do you have any suggestions on how to fix this issue? Or should I make some code changes in the apps?